File: frontend/src/types/session.ts
export interface Session {
  session_id: string;
  alias: string | null;
  ip_address: string;
  os_type: 'windows' | 'linux';
  shell_type: 'tcp' | 'hoaxshell' | 'netcat' | 'powershell';
  sibling_id: string | null;
  connected_at: string;
  alive: boolean;
}

export interface CommandLogEntry {
  id: number;
  session_id: string;
  command: string;
  output: string | null;
  flagged: boolean;
  executed_at: string;
}

export interface ExecResult {
  success: boolean;
  flagged: boolean;
  warning: string | null;
  message?: string;
}

File: backend/app/models/session.py
from pydantic import BaseModel
from datetime import datetime
from typing import Optional


class SessionInfo(BaseModel):
    session_id: str
    alias: Optional[str] = None
    ip_address: str
    os_type: str
    shell_type: str
    sibling_id: Optional[str] = None
    connected_at: datetime
    alive: bool = True

    class Config:
        from_attributes = True


class SessionExec(BaseModel):
    command: str


class SessionAlias(BaseModel):
    alias: str


class CommandLogEntry(BaseModel):
    id: int
    session_id: str
    command: str
    output: Optional[str] = None
    flagged: bool = False
    executed_at: datetime

    class Config:
        from_attributes = True
